core/extension/gdextension_manager.cpp

```diff
@@ -299,13 +299,17 @@ bool GDExtensionManager::ensure_extensions_loaded(const HashSet<String> &p_exten
 	}
 
 	Vector<String> loaded_extensions = get_loaded_extensions();
+	int i = 0;
 	for (const String &loaded_extension : loaded_extensions) {
 		if (!p_extensions.has(loaded_extension)) {
 			// The extension may not have a .gdextension file.
-			if (!FileAccess::exists(loaded_extension)) {
+			const Ref<GDExtension> extension = GDExtensionManager::get_singleton()->get_extension(loaded_extensions[i]);
+			if (!extension->get_loader()->is_library_existing()) {
 				extensions_removed.push_back(loaded_extension);
 			}
 		}
+
+		i++;
```

Comment:
Why keep track of the index i here? You can use `loaded_extension` instead of `loaded_extensions[i]`

Reply:
Thanks for the review, I don't know how I missed it :)